src/java.base/share/classes/sun/net/www/http/KeepAliveCache.java

```diff
@@ -101,50 +107,65 @@ static int getMaxConnections() {
 
     static final int LIFETIME = 5000;
 
+    // This class is never serialized (see writeObject/readObject).
+    private final ReentrantLock cacheLock = new ReentrantLock();
```

Comment:
This got introduced in JDK 16 with https://bugs.openjdk.org/browse/JDK-8229867, please don't introduce such changes in a backport. The patch needs to be done without JDK-8229867 being present in JDK 11u